### PR TITLE
fix update ScaledObject

### LINKED ISSUE #158: Error updating ScaleObject: the object has been modified
I get this error when trying to scale active ScaleObject

```
time="2019-05-07T20:02:31Z" level=error msg="Error updating scaledObject (hpa/azure-servicebus-queue-scaledobject) status: Operation cannot be fulfilled on scaledobjects.keda.k8s.io \"azure-servicebus-queue-scaledobject\": the object has been modified; please apply your changes to the latest version and try again"
```

The error is thrown here:
scale_handler.go:269
```go
func (h *ScaleHandler) updateScaledObject(scaledObject *keda_v1alpha1.ScaledObject) error {
	_, err := h.kedaClient.KedaV1alpha1().ScaledObjects(scaledObject.GetNamespace()).Update(scaledObject)
	if err != nil {
		log.Errorf("Error updating scaledObject (%s/%s) status: %s", scaledObject.GetNamespace(), scaledObject.GetName(), err.Error())
	}
	return err
}
```

### PR DESCRIPTION
**Pull request by aminebizid**
Fixes #158

### CODE REVIEW

File: pkg/handler/scale_handler.go [hunk 1]
```diff
@@ -259,18 +258,19 @@ func (h *ScaleHandler) scaleDeployment(deployment *apps_v1.Deployment, scaledObj
...
 	} else {
 		log.Debugf("deployment (%s/%s) no change", deployment.GetNamespace(), deployment.GetName())
 	}
+	return scaledObject
 }
 
-func (h *ScaleHandler) updateScaledObject(scaledObject *keda_v1alpha1.ScaledObject) error {
-	_, err := h.kedaClient.KedaV1alpha1().ScaledObjects(scaledObject.GetNamespace()).Update(scaledObject)
+func (h *ScaleHandler) updateScaledObject(scaledObject *keda_v1alpha1.ScaledObject) (*v1alpha1.ScaledObject, error) {
+	newscaledObject, err := h.kedaClient.KedaV1alpha1().ScaledObjects(scaledObject.GetNamespace()).Update(scaledObject)
```
Comment (yaron2) [inline pkg/handler/scale_handler.go]: If you assign the ```scaledObject``` parameter to the output of ```.Update()```, this is the only change needed as ```scaledObject``` is passed as a pointer and thus mutate its value without needing to explicitly return it from the all methods preceding ```updateScaledObject```.

Comment (aminebizid) [reply pkg/handler/scale_handler.go]: Done

**Comment by ahmelsayed** [comment]
Thanks for your contribution @aminebizid!

**Comment by yaron2** [comment]
lgtm

**Pull request merged**